src/system/SystemPacketBuffer.cpp

```diff
@@ -530,12 +530,19 @@ PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aRese
 
     CHIP_SYSTEM_FAULT_INJECT(FaultInjection::kFault_PacketBufferNew, return PacketBufferHandle());
 
-    // TODO: Change the max to a lower value
-    if (aAvailableSize > UINT32_MAX || lAllocSize > PacketBuffer::kMaxSizeWithoutReserve || lBlockSize > UINT32_MAX)
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
```

Comment:
So... this got missed in the PR that made this thing take `size_t` for `aAvailableSize`, but all the static_asserts and overflow checks in this function are now broken.  The comment on lines 511/512 no longer has any bearing on reality, etc.

Before we make more changes to this function, can we please fix it to actually be overflow-safe again?